docker-compose.yml

```diff
@@ -662,6 +662,35 @@ services:
         /arrow/ci/scripts/cpp_build.sh /arrow /build &&
         /arrow/ci/scripts/r_test.sh /arrow"
 
+  r-docker:
```

Comment:
`docker` is not meaningful in this context. We have an `ubuntu-r` and another `conda-r` build, this should align too. How does it differ? Could it replace either one?

According to your comments this should work on any base image with R installed, so we should capture that information somehow. Seemingly `rhub` provides the compatible base images, so perhaps we should call the docker service as `rhub-r` instead of `r-docker` and hardcode the `R_DOCKER_ORG` to `rhub`. 

I also find the three env variables a bit tedious, how about hardcoding `R_DOCKER_ORG` as I mentioned above and  merge `R_DOCKER_IMAGE` and `R_DOCKER_TAG` into `RHUB_IMAGE`?

Reply:
I would be ok with deleting the `ubuntu-r` job; technically that tests something slightly different (building the C++ lib and passing its location to R installation with `LIB_DIR`/`INCLUDE_DIR`) but honestly these new jobs test more things that we aren't covering elsewhere (operating systems, static builds)

I don't want to hard-code `rhub` because I'm not using only R images from the `rhub` organization. I'm already using images from both `rhub` and `rstudio`, and there are other Docker Hub orgs out there that we could draw from, if it turns out that we want/need to expand the build matrix.

I agree that the three env variables are tedious and would be happy to use a single `R_DOCKER=rhub/ubuntu-gcc-release:latest` var. But see L672 and 677: that would result in an invalid image name of `apache/arrow-dev:r-rhub/ubuntu-gcc-release:latest`. I considered a few alternatives and this seemed to be the least bad. Plus, it's consistent with the other jobs, which have 2-3 env vars that parametrize them. The defaults are set in .env, just like they are for `ARCH` and `UBUNTU` et al.

Reply:
Ok, how about naming it as `r` simply then instead of `r-docker` ? :) 

`docker-compose build r` vs `docker-compose build r-docker`